Kernel/Bus/USB/USBDevice.h

```diff
@@ -55,10 +60,13 @@ class Device : public RefCounted<Device> {
     u16 m_vendor_id { 0 };                      // This device's vendor ID assigned by the USB group
     u16 m_product_id { 0 };                     // This device's product ID assigned by the USB group
     USBDeviceDescriptor m_device_descriptor {}; // Device Descriptor obtained from USB Device
+    Vector<USBConfiguration> m_configurations;  // Configurations for this device
 
     NonnullRefPtr<USBController> m_controller;
     NonnullOwnPtr<Pipe> m_default_pipe; // Default communication pipe (endpoint0) used during enumeration
 
+    OwnPtr<USBConfiguration> m_current_configuration { nullptr };
```

Comment:
This is not used anywhere.
Also it's very suspicious that we'd have an owned, heap-allocated `USBConfiguration` here, but also a `Vector` or `USBConfiguration` values in `m_configurations`.
If you need to point at a `USBConfiguration` in the vector, an index would be the most appropriate thing to store. (A pointer would become invalid if/when the vector resizes.)

Reply:
This raises a question on whether we want/need a new data structure that is something in between `Array` and `Vector`, so you allocate a Vector once on the stack frame, and you pass it to a structure constructor that allocates that much of memory being needed, but you can't change the allocated memory count never in runtime after construction point, which suits great the fact that after the enumeration of a USB device configuration, you will never encounter a situation where you find new configurations so you basically don't need that capability after construction point.

Reply:
Well, it seems like `FixedArray` served this purpose for many things, so we might be able to use it here too?

Reply:
> Well, it seems like `FixedArray` served this purpose for many things, so we might be able to use it here too?

Not a bad idea. I'm effectively using `Vector` in this instance in the same way that we'd use a `FixedArray`, so it might be best to attempt an allocation of one via `TRY()`. As we go further down the chain, we also know the size of each of the arrays we'll require due to the descriptor telling us how many descriptors there are.